servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TestExecutor.java

```diff
@@ -314,18 +321,118 @@ private static boolean executeOne(Queue<RunnableWrapper> tasks, AtomicInteger ta
+                    try {
+                        asyncContextMapHolder.asyncContextMap(InvalidAsyncContextMap.INSTANCE);
+                        delegate.run();
+                        assertThat("ContextMap was not restored",
+                                asyncContextMapHolder.asyncContextMap(), sameInstance(InvalidAsyncContextMap.INSTANCE));
+                    } finally {
+                        asyncContextMapHolder.asyncContextMap(acmhPrev);
+                    }
+                } else {
+                    delegate.run();
+                }
+                assertThat("ContextMap was not restored",
+                        contextThreadLocal.get(), sameInstance(InvalidAsyncContextMap.INSTANCE));
+            } finally {
+                contextThreadLocal.set(tlPrev);
+                current.setName(oldName);
+            }
+        }
+    }
+
+    private static final class InvalidAsyncContextMap implements AsyncContextMap {
+        static final AsyncContextMap INSTANCE = new InvalidAsyncContextMap();
+
+        private InvalidAsyncContextMap() {
+            // singleton
+        }
+
+        @Nullable
+        @Override
+        public <T> T get(final Key<T> key) {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Override
+        public boolean containsKey(final Key<?> key) {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Override
+        public boolean isEmpty() {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Override
+        public int size() {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Nullable
+        @Override
+        public <T> T put(final Key<T> key, @Nullable final T value) {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Override
+        public void putAll(final Map<Key<?>, Object> map) {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Override
+        public <T> T remove(final Key<T> key) {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Override
+        public boolean removeAll(final Iterable<Key<?>> entries) {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Override
+        public void clear() {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Nullable
+        @Override
+        public Key<?> forEach(final BiPredicate<Key<?>, Object> consumer) {
+            throw new AssertionError("Invalid access of AsyncContextMap");
+        }
+
+        @Override
+        public AsyncContextMap copy() {
+            throw new AssertionError("Invalid access of AsyncContextMap");
```

Comment:
Consider moving the `throw new ...` into a private method to avoid duplicating the cause message.